src/differential_arithmetic.jl

```diff
@@ -32,15 +32,19 @@ Base.:*(::DoesNotExist, ::Zero) = Zero()
 Base.:*(::Zero, ::DoesNotExist) = Zero()
 
 
-Base.:+(::Zero, b::Zero) = Zero()
+Base.:+(::Zero, ::Zero) = Zero()
 Base.:*(::Zero, ::Zero) = Zero()
 for T in (:One, :AbstractThunk, :Any)
     @eval Base.:+(::Zero, b::$T) = b
     @eval Base.:+(a::$T, ::Zero) = a
 
-    @eval Base.:*(::Zero, ::$T) = Zero()
-    @eval Base.:*(::$T, ::Zero) = Zero()
+    if T !== :Any
```

Comment:
if we defined:

```
Base.zero(::AbstractDifferential) = Zero()
```
then we could remove this conditional.

I think that might be nice, what do you think?

Reply:
Yeah, that's pretty nice.